plugins/modules/vmware_vm_shell.py

```diff
@@ -210,7 +210,10 @@ def __init__(self, module):
         datacenter_name = module.params['datacenter']
         cluster_name = module.params['cluster']
         folder = module.params['folder']
-        self.pm = self.content.guestOperationsManager.processManager
+        try:
+            self.pm = self.content.guestOperationsManager.processManager
+        except vmodl.fault.ManagedObjectNotFound:
+            pass
```

Comment:
I would fail the module since the intended VM is deleted already.

Reply:
the exception raised is not necessary related to the intended VM. It looks like it may be raised for a deleted VM other than the intended one

Reply:
This issue is very similar to [mine](https://github.com/ansible-collections/community.vmware/pull/798). When iterating over all objects, the modules fail if an unrelated object is removed during execution. This makes it very difficult to have several tasks running at the same time if you are adding and removing items from vSphere inventory.